app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt

```diff
@@ -124,4 +142,31 @@ constructor(
   fun deauthenticate() {
     auth.signOut()
   }
+
+  suspend fun makeApiRequest(url: URL): String {
+    Log.d("AuthenticationController", "Making request to $url")
+
+    // refresh the access token if necessary
+    if (!refreshSpotifyToken()) {
+      Log.d("AuthenticationController", "Failed to refresh Spotify token")
+      return "FAILURE"
+    }
+    // Get the access token from the AuthTokenRepository
+    val accessToken =
+        tokenRepository.getAuthToken(AuthTokenRepository.AuthTokenType.SPOTIFY_ACCESS_TOKEN)
+
+    Log.d("AuthenticationController", "Access token: $accessToken")
+    return try {
+      (withContext(Dispatchers.IO) { url.openConnection() } as HttpURLConnection).run {
```

Comment:
Use the global scope like yoric did before no ?

Reply:
Yup, corrected it, thanks